### ISSUE TITLE
perf performance drop with master code compared with v18.10.x

### ISSUE THREAD

**Issue by wanqunintel** [spdk/spdk#624]
## Issue Summary
Perf performance drop compared with v18.10.x

## Expected Behavior
Performance should keep almost the same

## Current Behavior
Performance drops to 1/3 compared with v18.10.x

## Steps to Reproduce
1. git clone https://github.com/spdk/spdk.git 
2. git checkout remotes/origin/v18.10.x
3. git submodule update --init
4. ./configure --with-rdma
5. make -j
6. On target, running with the command 
./app/nvmf_tgt/nvmf_tgt -c /root/nvmf_single_nvme_rdma.conf
file nvmf_single_nvme_rdma.conf content as follows
[Global]
[Bdev]
[Pmem]
Blk /path/to/pmem-pool Pmem0
[Nvmf]
AcceptorPollRate 10000
[Transport]
Type RDMA
MaxQueueDepth 128
[Nvme]
TransportID "trtype:PCIe traddr:0000:d8:00.0" Nvme0
[Subsystem1]
NQN nqn.2016-06.io.spdk:cnode1
Listen RDMA 192.168.3.11:4420
AllowAnyHost Yes
Host nqn.2016-06.io.spdk:init
SN SPDK00000000000001
MaxNamespaces 20
Namespace Nvme0n1 1
7. on initiator, running following command
./examples/nvme/perf/perf -r 'trtype:RDMA adrfam:IPv4 traddr:192.168.3.1 trsvcid:4420' -s 1024 -q 10 -o 16384 -w randrw -M 50 -t 9
Starting SPDK v18.10.2-pre / DPDK 18.08.0 initialization...
[ DPDK EAL parameters: perf --no-shconf -c 0x1 -m 1024 --no-pci --file-prefix=spdk_pid40188 ]
EAL: Detected 48 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
Initializing NVMe Controllers
Attaching to NVMe over Fabrics controller at 192.168.3.1:4420: nqn.2016-06.io.spdk:cnode1
Attached to NVMe over Fabrics controller at 192.168.3.1:4420: nqn.2016-06.io.spdk:cnode1
Associating SPDK bdev Controller (SPDK00000000000001  ) with lcore 0
Initialization complete. Launching workers.
Starting thread on core 0
========================================================
                                                                                            Latency(us)
Device Information                                     :       IOPS       MB/s    Average        min        max
SPDK bdev Controller (SPDK00000000000001  ) from core 0:  143291.90    2238.94      69.76      14.61    8530.24
========================================================
Total                                                  :  143291.90    2238.94      69.76      14.61    8530.24
8. Change to master and rerun the command, on target, the configure file as follows
 [global]
 [Bdev]
 [Pmem]
  Blk /path/to/pmem-pool Pmem0
   [Nvmf]
   AcceptorPollRate 10000
   [Transport]
   Type RDMA
  MaxQueueDepth 128
  NumSharedBuffers 2048
  [Nvme]
  TransportID "trtype:PCIe traddr:0000:82:00.0" Nvme0
  [Subsystem1]
  NQN nqn.2016-06.io.spdk:cnode1
  Listen RDMA 192.168.3.1:4420
  AllowAnyHost Yes
  Host nqn.2016-06.io.spdk:init
  SN SPDK00000000000001
  MaxNamespaces 20
  Namespace Nvme0n1 1
On initiator, run the same command and got result as follows
./examples/nvme/perf/perf -r 'trtype:RDMA adrfam:IPv4 traddr:192.168.3.1 trsvcid:4420' -s 1024 -q 10 -o 16384 -w randrw -M 50 -t 9
Starting SPDK v19.01-pre / DPDK 18.08.0 initialization...
[ DPDK EAL parameters: perf --no-shconf -c 0x1 -m 1024 --no-pci --base-virtaddr=0x200000000000 --file-prefix=spdk_pid40199 ]
EAL: Detected 48 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
Initializing NVMe Controllers
Attaching to NVMe over Fabrics controller at 192.168.3.1:4420: nqn.2016-06.io.spdk:cnode1
Attached to NVMe over Fabrics controller at 192.168.3.1:4420: nqn.2016-06.io.spdk:cnode1
Associating SPDK bdev Controller (SPDK00000000000001  ) with lcore 0
Initialization complete. Launching workers.
Starting thread on core 0
========================================================
                                                                                            Latency(us)
Device Information                                     :       IOPS       MB/s    Average        min        max
SPDK bdev Controller (SPDK00000000000001  ) from core 0:   50225.00     784.77     199.07      17.85    7716.42
========================================================
Total                                                  :   50225.00     784.77     199.07      17.85    7716.42

**Comment by wanqunintel** [comment]
use malloc as backend and retested with no big delta with master and v18.10.x

**Comment by changpe1** [comment]
@wanqunintel Can you also post the performance result using kernel initiator driver with fio ?

**Comment by optimistyzy** [comment]
@wanqunintel  As listed by @changpe1 , Can you paste some performance comparison data?

**Comment by wanqunintel** [comment]
@optimistyzy @changpe1 Just retested with 10 times on different spdk version and it shows following result, it seems there are some fix for it in the master, is there anyone know which patch has fixed this issue?
**18.10**
Starting SPDK v18.10.2-pre / DPDK 18.08.0 initialization...
[ DPDK EAL parameters: perf --no-shconf -c 0x1 -m 1024 --no-pci --file-prefix=spdk_pid43909 ]
EAL: Detected 48 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
Initializing NVMe Controllers
Attaching to NVMe over Fabrics controller at 192.168.3.1:4420: nqn.2016-06.io.spdk:cnode1
Attached to NVMe over Fabrics controller at 192.168.3.1:4420: nqn.2016-06.io.spdk:cnode1
Associating SPDK bdev Controller (SPDK00000000000001  ) with lcore 0
Initialization complete. Launching workers.
Starting thread on core 0
========================================================
                                                                                            Latency(us)
Device Information                                     :       IOPS       MB/s    Average        min        max
SPDK bdev Controller (SPDK00000000000001  ) from core 0:  143862.00    2247.84      69.49      14.65    7697.31
========================================================
Total                                                  :  143862.00    2247.84      69.49      14.65    7697.31

**19.01**
Starting SPDK v19.01 / DPDK 18.11.0 initialization...
[ DPDK EAL parameters: perf --no-shconf -c 0x1 -m 1024 --no-pci --base-virtaddr=0x200000000000 --file-prefix=spdk_pid44988 ]
EAL: Detected 48 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
Initializing NVMe Controllers
Attaching to NVMe over Fabrics controller at 192.168.3.1:4420: nqn.2016-06.io.spdk:cnode1
Attached to NVMe over Fabrics controller at 192.168.3.1:4420: nqn.2016-06.io.spdk:cnode1
Associating SPDK bdev Controller (SPDK00000000000001  ) with lcore 0
Initialization complete. Launching workers.
Starting thread on core 0
========================================================
                                                                                            Latency(us)
Device Information                                     :       IOPS       MB/s    Average        min        max
SPDK bdev Controller (SPDK00000000000001  ) from core 0:   44933.67     702.09     222.51      18.00    7477.98
========================================================
Total                                                  :   44933.67     702.09     222.51      18.00    7477.98

**Master (commit 55be9a57a619e275826314e90e8c7c1d6348abc7)** 
Starting SPDK v19.04-pre / DPDK 18.11.0 initialization...
[ DPDK EAL parameters: perf --no-shconf -c 0x1 -m 1024 --no-pci --base-virtaddr=0x200000000000 --file-prefix=spdk_pid45548 ]
EAL: Detected 48 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
Initializing NVMe Controllers
Attaching to NVMe over Fabrics controller at 192.168.3.1:4420: nqn.2016-06.io.spdk:cnode1
Attached to NVMe over Fabrics controller at 192.168.3.1:4420: nqn.2016-06.io.spdk:cnode1
Associating SPDK bdev Controller (SPDK00000000000001  ) with lcore 0
Initialization complete. Launching workers.
Starting thread on core 0
========================================================
                                                                                            Latency(us)
Device Information                                     :       IOPS       MB/s    Average        min        max
SPDK bdev Controller (SPDK00000000000001  ) from core 0:  143217.89    2237.78      69.79      14.68    7713.35
========================================================
Total                                                  :  143217.89    2237.78      69.79      14.68    7713.35

**Comment by optimistyzy** [comment]
@wanqunintel  So can we conclude that this issue is fixed in the master branch.

**Comment by wanqunintel** [comment]
@optimistyzy it seems fix, but we are not sure what's the issue there, i discussed with changpeng and better to understand the reason how it's fixed. So i put the but scrub there to get some info.

**Comment by wanqunintel** [comment]
@changpe1 , to get more info for the debuging if perf issue or nvme driver issue, fio result is attached as follows, config file for fio job is as follow
[global]
invalidate=1
norandommap=1
thread=1
rw=randrw
runtime=10
ioengine=libaio
direct=1
#bs=4096
bs=16384
size=1G
#iodepth=128
iodepth=10
group_reporting
#numjobs=32
time_based=1
[job0]
filename=/dev/nvme0n1
**1810**
job0: (g=0): rw=randrw, bs=(R) 16.0KiB-16.0KiB, (W) 16.0KiB-16.0KiB, (T) 16.0KiB-16.0KiB, ioengine=libaio, iodepth=10
fio-3.6-34-g5eac
Starting 1 thread

job0: (groupid=0, jobs=1): err= 0: pid=2310: Wed Feb 13 10:48:00 2019
   read: IOPS=20.8k, BW=325MiB/s (341MB/s)(3251MiB/10003msec)
    slat (usec): min=3, max=167, avg= 7.61, stdev= 2.95
    clat (usec): min=84, max=7779, avg=416.95, stdev=548.38
     lat (usec): min=90, max=7791, avg=424.67, stdev=548.37
    clat percentiles (usec):
     |  1.00th=[  129],  5.00th=[  141], 10.00th=[  149], 20.00th=[  161],
     | 30.00th=[  174], 40.00th=[  186], 50.00th=[  208], 60.00th=[  251],
     | 70.00th=[  314], 80.00th=[  379], 90.00th=[  930], 95.00th=[ 1926],
     | 99.00th=[ 2638], 99.50th=[ 2737], 99.90th=[ 2933], 99.95th=[ 3032],
     | 99.99th=[ 6652]
   bw (  KiB/s): min=322144, max=346336, per=100.00%, avg=332878.40, stdev=5660.54, samples=20
   iops        : min=20134, max=21646, avg=20804.90, stdev=353.78, samples=20
  write: IOPS=20.8k, BW=325MiB/s (341MB/s)(3249MiB/10003msec)
    slat (nsec): min=3427, max=50193, avg=8720.37, stdev=3054.69
    clat (usec): min=17, max=6066, avg=45.64, stdev=48.51
     lat (usec): min=34, max=6070, avg=54.46, stdev=48.66
    clat percentiles (usec):
     |  1.00th=[   33],  5.00th=[   35], 10.00th=[   36], 20.00th=[   37],
     | 30.00th=[   39], 40.00th=[   40], 50.00th=[   42], 60.00th=[   45],
     | 70.00th=[   48], 80.00th=[   53], 90.00th=[   61], 95.00th=[   68],
     | 99.00th=[   84], 99.50th=[   91], 99.90th=[  112], 99.95th=[  131],
     | 99.99th=[  676]
   bw (  KiB/s): min=324384, max=347648, per=100.00%, avg=332734.40, stdev=6139.72, samples=20
   iops        : min=20274, max=21728, avg=20795.90, stdev=383.73, samples=20
  lat (usec)   : 20=0.01%, 50=37.30%, 100=12.59%, 250=29.96%, 500=12.43%
  lat (usec)   : 750=1.74%, 1000=1.21%
  lat (msec)   : 2=2.46%, 4=2.29%, 10=0.02%
  cpu          : usr=8.57%, sys=28.42%, ctx=184714, majf=0, minf=104
  IO depths    : 1=0.1%, 2=0.1%, 4=0.1%, 8=100.0%, 16=0.0%, 32=0.0%, >=64=0.0%
     submit    : 0=0.0%, 4=100.0%, 8=0.0%, 16=0.0%, 32=0.0%, 64=0.0%, >=64=0.0%
     complete  : 0=0.0%, 4=100.0%, 8=0.0%, 16=0.1%, 32=0.0%, 64=0.0%, >=64=0.0%
     issued rwts: total=208056,207961,0,0 short=0,0,0,0 dropped=0,0,0,0
     latency   : target=0, window=0, percentile=100.00%, depth=10

Run status group 0 (all jobs):
   READ: bw=325MiB/s (341MB/s), 325MiB/s-325MiB/s (341MB/s-341MB/s), io=3251MiB (3409MB), run=10003-10003msec
  WRITE: bw=325MiB/s (341MB/s), 325MiB/s-325MiB/s (341MB/s-341MB/s), io=3249MiB (3407MB), run=10003-10003msec

Disk stats (read/write):
  nvme0n1: ios=0/0, merge=0/0, ticks=0/0, in_queue=0, util=0.00%
**1901**
job0: (g=0): rw=randrw, bs=(R) 16.0KiB-16.0KiB, (W) 16.0KiB-16.0KiB, (T) 16.0KiB-16.0KiB, ioengine=libaio, iodepth=10
fio-3.6-34-g5eac
Starting 1 thread

job0: (groupid=0, jobs=1): err= 0: pid=2922: Wed Feb 13 10:51:27 2019
   read: IOPS=23.0k, BW=360MiB/s (377MB/s)(3600MiB/10001msec)
    slat (usec): min=3, max=166, avg= 7.49, stdev= 1.54
    clat (usec): min=65, max=8027, avg=374.71, stdev=507.91
     lat (usec): min=92, max=8033, avg=382.30, stdev=507.88
    clat percentiles (usec):
     |  1.00th=[  126],  5.00th=[  137], 10.00th=[  145], 20.00th=[  155],
     | 30.00th=[  165], 40.00th=[  176], 50.00th=[  192], 60.00th=[  225],
     | 70.00th=[  285], 80.00th=[  343], 90.00th=[  758], 95.00th=[ 1713],
     | 99.00th=[ 2606], 99.50th=[ 2704], 99.90th=[ 2868], 99.95th=[ 2966],
     | 99.99th=[ 5800]
   bw (  KiB/s): min=361984, max=377280, per=100.00%, avg=368702.32, stdev=5093.51, samples=19
   iops        : min=22624, max=23580, avg=23043.89, stdev=318.34, samples=19
  write: IOPS=23.0k, BW=360MiB/s (378MB/s)(3601MiB/10001msec)
    slat (nsec): min=3471, max=38125, avg=8454.19, stdev=1598.21
    clat (usec): min=23, max=5434, avg=41.72, stdev=37.50
     lat (usec): min=33, max=5442, avg=50.26, stdev=37.56
    clat percentiles (usec):
     |  1.00th=[   31],  5.00th=[   34], 10.00th=[   35], 20.00th=[   36],
     | 30.00th=[   37], 40.00th=[   38], 50.00th=[   39], 60.00th=[   41],
     | 70.00th=[   44], 80.00th=[   48], 90.00th=[   53], 95.00th=[   58],
     | 99.00th=[   70], 99.50th=[   75], 99.90th=[   87], 99.95th=[   93],
     | 99.99th=[  180]
   bw (  KiB/s): min=359136, max=378592, per=99.99%, avg=368613.05, stdev=4668.37, samples=19
   iops        : min=22446, max=23662, avg=23038.32, stdev=291.77, samples=19
  lat (usec)   : 50=42.92%, 100=7.07%, 250=32.41%, 500=11.25%, 750=1.30%
  lat (usec)   : 1000=1.08%
  lat (msec)   : 2=2.05%, 4=1.91%, 10=0.01%
  cpu          : usr=12.22%, sys=43.30%, ctx=197332, majf=0, minf=104
  IO depths    : 1=0.1%, 2=0.1%, 4=0.1%, 8=100.0%, 16=0.0%, 32=0.0%, >=64=0.0%
     submit    : 0=0.0%, 4=100.0%, 8=0.0%, 16=0.0%, 32=0.0%, 64=0.0%, >=64=0.0%
     complete  : 0=0.0%, 4=100.0%, 8=0.0%, 16=0.1%, 32=0.0%, 64=0.0%, >=64=0.0%
     issued rwts: total=230397,230441,0,0 short=0,0,0,0 dropped=0,0,0,0
     latency   : target=0, window=0, percentile=100.00%, depth=10

Run status group 0 (all jobs):
   READ: bw=360MiB/s (377MB/s), 360MiB/s-360MiB/s (377MB/s-377MB/s), io=3600MiB (3775MB), run=10001-10001msec
  WRITE: bw=360MiB/s (378MB/s), 360MiB/s-360MiB/s (378MB/s-378MB/s), io=3601MiB (3776MB), run=10001-10001msec

Disk stats (read/write):
  nvme0n1: ios=0/0, merge=0/0, ticks=0/0, in_queue=0, util=0.00%
**Master**
job0: (g=0): rw=randrw, bs=(R) 16.0KiB-16.0KiB, (W) 16.0KiB-16.0KiB, (T) 16.0KiB-16.0KiB, ioengine=libaio, iodepth=10
fio-3.6-34-g5eac
Starting 1 thread

job0: (groupid=0, jobs=1): err= 0: pid=3494: Wed Feb 13 10:54:30 2019
   read: IOPS=23.0k, BW=360MiB/s (377MB/s)(3600MiB/10001msec)
    slat (usec): min=3, max=151, avg= 6.88, stdev= 2.79
    clat (usec): min=81, max=8085, avg=374.18, stdev=502.46
     lat (usec): min=91, max=8090, avg=381.16, stdev=502.43
    clat percentiles (usec):
     |  1.00th=[  127],  5.00th=[  139], 10.00th=[  145], 20.00th=[  157],
     | 30.00th=[  167], 40.00th=[  178], 50.00th=[  194], 60.00th=[  227],
     | 70.00th=[  285], 80.00th=[  343], 90.00th=[  750], 95.00th=[ 1696],
     | 99.00th=[ 2573], 99.50th=[ 2704], 99.90th=[ 2868], 99.95th=[ 2966],
     | 99.99th=[ 5604]
   bw (  KiB/s): min=360384, max=381664, per=100.00%, avg=368673.60, stdev=5062.17, samples=20
   iops        : min=22524, max=23854, avg=23042.10, stdev=316.39, samples=20
  write: IOPS=23.0k, BW=360MiB/s (378MB/s)(3601MiB/10001msec)
    slat (usec): min=3, max=134, avg= 8.03, stdev= 2.93
    clat (usec): min=17, max=5360, avg=43.22, stdev=43.65
     lat (usec): min=31, max=5370, avg=51.35, stdev=43.84
    clat percentiles (usec):
     |  1.00th=[   31],  5.00th=[   34], 10.00th=[   35], 20.00th=[   36],
     | 30.00th=[   37], 40.00th=[   38], 50.00th=[   40], 60.00th=[   42],
     | 70.00th=[   46], 80.00th=[   50], 90.00th=[   57], 95.00th=[   64],
     | 99.00th=[   80], 99.50th=[   87], 99.90th=[  101], 99.95th=[  109],
     | 99.99th=[  155]
   bw (  KiB/s): min=360160, max=374144, per=100.00%, avg=368771.20, stdev=3716.99, samples=20
   iops        : min=22510, max=23384, avg=23048.20, stdev=232.38, samples=20
  lat (usec)   : 20=0.01%, 50=40.39%, 100=9.57%, 250=32.31%, 500=11.44%
  lat (usec)   : 750=1.29%, 1000=1.05%
  lat (msec)   : 2=2.06%, 4=1.88%, 10=0.01%
  cpu          : usr=8.79%, sys=28.33%, ctx=202866, majf=0, minf=104
  IO depths    : 1=0.1%, 2=0.1%, 4=0.1%, 8=100.0%, 16=0.0%, 32=0.0%, >=64=0.0%
     submit    : 0=0.0%, 4=100.0%, 8=0.0%, 16=0.0%, 32=0.0%, 64=0.0%, >=64=0.0%
     complete  : 0=0.0%, 4=100.0%, 8=0.0%, 16=0.1%, 32=0.0%, 64=0.0%, >=64=0.0%
     issued rwts: total=230429,230483,0,0 short=0,0,0,0 dropped=0,0,0,0
     latency   : target=0, window=0, percentile=100.00%, depth=10

Run status group 0 (all jobs):
   READ: bw=360MiB/s (377MB/s), 360MiB/s-360MiB/s (377MB/s-377MB/s), io=3600MiB (3775MB), run=10001-10001msec
  WRITE: bw=360MiB/s (378MB/s), 360MiB/s-360MiB/s (378MB/s-378MB/s), io=3601MiB (3776MB), run=10001-10001msec

Disk stats (read/write):
  nvme0n1: ios=0/0, merge=0/0, ticks=0/0, in_queue=0, util=0.00%

**Comment by Smalinuxer** [comment]
finally . i find out what the problem it is.
so i do this test in version 6f62f0a187341a9e1a136d2080cb11dae65cb6ed (which is between master and v18.10.x)

* git clone https://github.com/spdk/spdk.git
* git checkout 6f62f0a187341a9e1a136d2080cb11dae65cb6ed 
* git submodule update --init
* ./configure --with-rdma
* make -j
* ./examples/nvme/perf/perf -r 'trtype:RDMA adrfam:IPv4 traddr:192.168.91.11 trsvcid:4420' -s 1024 -q 10 -o 16384 -w randrw -M 50 -t 9

the result same as the @wanqunintel

```

Starting SPDK v19.01-pre / DPDK 18.11.0 initialization...
[ DPDK EAL parameters: perf --no-shconf -c 0x1 -m 1024 --no-pci --base-virtaddr=0x200000000000 --file-prefix=spdk_pid40265 ]
EAL: Detected 72 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
Initializing NVMe Controllers
Attaching to NVMe over Fabrics controller at 192.168.91.11:4420: nqn.2016-06.io.spdk:cnode1
Attached to NVMe over Fabrics controller at 192.168.91.11:4420: nqn.2016-06.io.spdk:cnode1
Associating SPDK bdev Controller (SPDK00000000000001  ) with lcore 0
Initialization complete. Launching workers.
Starting thread on core 0
========================================================
                                                                                            Latency(us)
Device Information                                     :       IOPS       MB/s    Average        min        max
SPDK bdev Controller (SPDK00000000000001  ) from core 0:   48156.78     752.45     207.54      24.04    7303.78
========================================================
Total                                                  :   48156.78     752.45     207.54      24.04    7303.78

```

and then i change the perf.c in line 590:

```
uint32_t max_xfer_size, entries, ns_size, sector_size;
```

to 

```
uint32_t max_xfer_size, entries,  sector_size;
uint64_t ns_size;
```

then make and run perf again.

```
Starting SPDK v19.01-pre / DPDK 18.11.0 initialization...
[ DPDK EAL parameters: perf --no-shconf -c 0x1 -m 1024 --no-pci --base-virtaddr=0x200000000000 --file-prefix=spdk_pid7056 ]
EAL: Detected 72 lcore(s)
EAL: Detected 2 NUMA nodes
EAL: No free hugepages reported in hugepages-1048576kB
EAL: Probing VFIO support...
Initializing NVMe Controllers
Attaching to NVMe over Fabrics controller at 192.168.91.11:4420: nqn.2016-06.io.spdk:cnode1
Attached to NVMe over Fabrics controller at 192.168.91.11:4420: nqn.2016-06.io.spdk:cnode1
Associating SPDK bdev Controller (SPDK00000000000001  ) with lcore 0
Initialization complete. Launching workers.
Starting thread on core 0
========================================================
                                                                                            Latency(us)
Device Information                                     :       IOPS       MB/s    Average        min        max
SPDK bdev Controller (SPDK00000000000001  ) from core 0:  132385.11    2068.52      75.43      24.43    5622.19
========================================================
Total                                                  :  132385.11    2068.52      75.43      24.43    5622.19

```

and it work normal .

**and this change have been patched.**

so close this issue pls.

**Comment by wanqunintel** [comment]
thanks, Jiaqi, after checking the log, this issue was fixed with this patch, bdc96b4dd623a844ba9e296837e854f5096bb35e, and introduced with this patch "f24629092489d70e64826c2bba116514efb1bacf" as described in the commit message
